dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/OssOperator.java

```diff
@@ -0,0 +1,321 @@
+
+        ossClient = buildOssClient();
+        ensureBucketSuccessfullyCreated(BUCKET_NAME);
+    }
+
+    @Override
+    public void close() throws IOException {
+        ossClient.shutdown();
+    }
+
+    @Override
+    public void createTenantDirIfNotExists(String tenantCode) throws Exception {
+        mkdir(tenantCode, getOssResDir(tenantCode));
+        mkdir(tenantCode, getOssUdfDir(tenantCode));
+    }
+
+    @Override
+    public String getResDir(String tenantCode) {
+        return getOssResDir(tenantCode) + FOLDER_SEPARATOR;
+    }
+
+    @Override
+    public String getUdfDir(String tenantCode) {
+        return getOssUdfDir(tenantCode) + FOLDER_SEPARATOR;
+    }
+
+    @Override
+    public boolean mkdir(String tenantCode, String path) throws IOException {
+        final String key = path + FOLDER_SEPARATOR;
+        if (!ossClient.doesObjectExist(BUCKET_NAME, key)) {
+            createOssPrefix(BUCKET_NAME, key);
+        }
+        return true;
+    }
+
+    protected void createOssPrefix(final String bucketName, final String key) {
+        ObjectMetadata metadata = new ObjectMetadata();
+        metadata.setContentLength(0);
+        InputStream emptyContent = new ByteArrayInputStream(new byte[0]);
+        PutObjectRequest putObjectRequest = new PutObjectRequest(bucketName, key, emptyContent, metadata);
+        ossClient.putObject(putObjectRequest);
+    }
+
+    @Override
+    public String getResourceFileName(String tenantCode, String fileName) {
+        if (fileName.startsWith(FOLDER_SEPARATOR)) {
+            fileName = fileName.replaceFirst(FOLDER_SEPARATOR, "");
+        }
+        return String.format(FORMAT_S_S, getOssResDir(tenantCode), fileName);
+    }
+
+    @Override
+    public String getFileName(ResourceType resourceType, String tenantCode, String fileName) {
+        if (fileName.startsWith(FOLDER_SEPARATOR)) {
+            fileName = fileName.replaceFirst(FOLDER_SEPARATOR, "");
+        }
+        return getDir(resourceType, tenantCode) + fileName;
+    }
+
+    @Override
+    public void download(String tenantCode, String srcFilePath, String dstFilePath, boolean deleteSource,
+                         boolean overwrite) throws IOException {
+        File dstFile = new File(dstFilePath);
+        if (dstFile.isDirectory()) {
+            Files.delete(dstFile.toPath());
+        } else {
+            Files.createDirectories(dstFile.getParentFile().toPath());
+        }
+        OSSObject ossObject = ossClient.getObject(BUCKET_NAME, srcFilePath);
+        try (
+                InputStream ossInputStream = ossObject.getObjectContent();
+                FileOutputStream fos = new FileOutputStream(dstFilePath)) {
+            byte[] readBuf = new byte[1024];
+            int readLen;
+            while ((readLen = ossInputStream.read(readBuf)) > 0) {
+                fos.write(readBuf, 0, readLen);
+            }
+        } catch (OSSException e) {
+            throw new IOException(e.getMessage());
```

Comment:
```suggestion
            throw new IOException(e);
```
Don't hide the stack.

Reply:
Sure, will fix it.